File: BorderlessApp/Borderless.ServiceLayer/Controllers/TranslationsController.cs
using System;
using System.Collections.Generic;
using System.Web.Http;
using System.Web.Http.Cors;
using Borderless.BusinessLayer;
using Borderless.Model.Entities;
using Borderless.ServiceLayer.Helpers;

namespace Borderless.ServiceLayer.Controllers
{
    [RoutePrefix("api")]
    [EnableCors(origins: "*", headers: "*", methods: "*")]
    public class TranslationsController : ApiController
    {
        private BLContext _context = new BLContext();

        [HttpGet]
        [Authorize]
        [Route("translations/{id:guid}")]
        public Translation GetById(Guid id)
        {
            return _context.Translations.GetById(id);
        }

        [HttpGet]
        [Authorize]
        [Route("phrases/{id:guid}/translations")]
        public List<Translation> GetAllByPhraseId(Guid phraseId)
        {
            return _context.Translations.GetAllByPhraseId(phraseId);
        }

        [HttpGet]
        [Authorize]
        [Route("phrases/{phraseId:guid}/translations/{languageId:guid}")]
        public List<Translation> GetAllByPhraseIdAndLanguageId(Guid phraseId, Guid languageId)
        {
            return _context.Translations.GetAllByPhraseIdAndLanguageId(phraseId, languageId);
        }

        [HttpGet]
        [Authorize]
        [Route("users/{id:guid}/translations")]
        public List<Translation> GetAllByUserId(Guid id)
        {
            return _context.Translations.GetAllByUserId(id);
        }

        [HttpPost]
        [Authorize]
        [Route("translations")]
        public Translation Add([FromBody]Translation translation)
        {
            Guid authenticatedUserId = ClaimsHelper.GetUserIdFromClaims();
            return _context.Translations.Add(translation, authenticatedUserId);
        }

        [HttpPut]
        [Authorize]
        [Route("translations/{id:guid}")]
        public Translation UpdateById(Guid id, [FromBody]Translation translation)
        {
            Guid authenticatedUserId = ClaimsHelper.GetUserIdFromClaims();
            return _context.Translations.UpdateById(id, translation, authenticatedUserId);
        }

        [HttpDelete]
        [Authorize]
        [Route("translations/{id:guid}")]
        public void DeleteById(Guid id)
        {
            Guid authenticatedUserId = ClaimsHelper.GetUserIdFromClaims();
            _context.Translations.DeleteById(id, authenticatedUserId);
        }
    }
}

File: BorderlessFrontEnd/scripts/controllers/translationsController.js
import { translationsService } from '../services/translationsService.js';
import { validator } from './validators/validator.js';

export const translationsController = (() => {
	async function getAllByPhraseId(phraseId) {
		validator.validateGuid(phraseId);
		return await translationsService.getAllByPhraseId(phraseId);
	}

	async function getAllByUserId(userId) {
		validator.validateGuid(userId);
		return await translationsService.getAllByUserId(userId);
	}

	async function getAllByPhraseIdAndLanguageId(phraseId, languageId) {
		validator.validateGuid(phraseId);
		validator.validateGuid(languageId);
		return await translationsService.getAllByPhraseIdAndLanguageId(phraseId, languageId);
	}

	async function getUserRoleByTranslationId(id) {
		validator.validateGuid(id);
		return await translationsService.getUserRoleByTranslationId(id);
	}

	async function getById(id) {
		validator.validateGuid(id);
		return await translationsService.getById(id);
	}

	async function add(text, phraseId, languageId) {
		validator.validateGuid(phraseId);
		validator.validateGuid(languageId);
		validator.validateTranslationText(text);
		return await translationsService.add(text, phraseId, languageId);
	}

	async function updateById(id, text) {
		validator.validateGuid(id);
		validator.validateTranslationText(text);
		return await translationsService.updateById(id, text);
	}

	async function deleteById(id) {
		validator.validateGuid(id);
		return await translationsService.deleteById(id);
	}

	return {
		getAllByPhraseId,
		getAllByUserId,
		getAllByPhraseIdAndLanguageId,
		getUserRoleByTranslationId,
		getById,
		add,
		updateById,
		deleteById,
	};
})();
